lib/Immediate/Immediate.cpp

```diff
@@ -197,9 +171,27 @@ bool swift::immediate::linkLLVMModules(llvm::Module *Module,
   llvm::LLVMContext &Ctx = SubModule->getContext();
   auto OldHandler = Ctx.getDiagnosticHandler();
   void *OldDiagnosticContext = Ctx.getDiagnosticContext();
-  Ctx.setDiagnosticHandler(linkerDiagnosticHandler, nullptr);
+
+  struct LinkerDiagnosticHandler : public llvm::DiagnosticHandler {
+    bool handleDiagnostics(const llvm::DiagnosticInfo &DI) override {
+      if (DI.getSeverity() != llvm::DS_Error)
+        return false;
+
+      std::string MsgStorage;
+      {
+        llvm::raw_string_ostream Stream(MsgStorage);
+        llvm::DiagnosticPrinterRawOStream DP(Stream);
+        DI.print(DP);
+      }
+      llvm::errs() << "Error linking swift modules\n";
+      llvm::errs() << MsgStorage << "\n";
+      return true;
+    }
+  };
+
+  Ctx.setDiagnosticHandler(std::unique_ptr<LinkerDiagnosticHandler>(), nullptr);
```

Comment:
Did you mean `llvm::makeUnique` here? This will create a null unique_ptr.

Reply:
Yes, sorry, I had to run and didn't notice that...